log/enabled.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package log // import "go.opentelemetry.io/otel/log"
+
+// EnabledParameters represents payload for [Logger]'s Enabled method.
+type EnabledParameters struct {
+	severity Severity
+}
+
+// Severity returns the [Severity] level.
+func (r *EnabledParameters) Severity() Severity {
+	return r.severity
+}
+
+// SetSeverity sets the [Severity] level.
+func (r *EnabledParameters) SetSeverity(level Severity) {
+	r.severity = level
+}
```

Comment:
```suggestion
// EnabledParameters represents payload for [Logger]'s Enabled method.
type EnabledParameters struct {
	severity Severity
	severitySet bool
}

// Severity returns the [Severity] level.
func (r *EnabledParameters) Severity() (Severity, bool) {
	return r.severity, r.severitySet
}

// SetSeverity sets the [Severity] level.
func (r *EnabledParameters) SetSeverity(level Severity) {
	r.severity = level
	r.severitySet = true
}
```

Reply:
I was thinking about the same 👍

Reply:
Yeah, I think if we are going to go the getter/setter route, we should leverage this type of interface.

Reply:
Addressed https://github.com/open-telemetry/opentelemetry-go/pull/5791/commits/9db67acf75c934899a286d017b8b44b538013679

Doc comments inspired by https://pkg.go.dev/sync#Map.Load